kong/init.lua

```diff
@@ -545,10 +520,16 @@ function Kong.init()
   end
 
   if config.database == "off" then
-    local err
-    declarative_entities, err, declarative_meta = parse_declarative_config(kong.configuration)
-    if not declarative_entities then
-      error(err)
+    if is_http_module or
+       (#config.proxy_listeners == 0 and
+        #config.admin_listeners == 0 and
+        #config.status_listeners == 0)
```

Comment:
I don't understand the logic of this `if`. If "is http" or ..."is stream but listening to nothing"? What is this testing, really?

Reply:
This achieves the following:

1. If HTTP is enabled, then config is loaded by the HTTP submodule and Stream submodule will not load any config
2. If HTTP is not enabled, then Stream subsystem will have to do the config loading itself

This is different from shdict based implementation as LMDB is shared between HTTP and Stream, so loading only need to happen once.

Reply:
Resolved.

Reply:
Answered in private meeting. This just reflects the conditions present on the template. It might be better if they were abstracted on a helper function and used here and in the template.